tests/python/gaia-ui-tests/gaiatest/apps/email/regions/setup.py

```diff
@@ -182,7 +183,14 @@ def tap_next(self):
                 *self._account_prefs_next_locator))))
 
     def check_for_emails_interval(self, value):
-        self.marionette.execute_script('document.querySelector("[data-l10n-id = settings-check-every-5min]").value = "%s";' % value)
+        # The following pref change allows us to check the mail within 1 second or longer,
+        # rather than the default value of 100 seconds
+        # The UI data layer of the UI is changed, because the minimum check mail time value is 5 min,
+        # which is far too long to check for in a test. This allows us to check earlier
+        self.marionette.execute_script("""
+            SpecialPowers.setIntPref('dom.requestSync.minInterval', 1);
+            document.querySelector("[data-l10n-id = settings-check-every-5min]").value = '%s';
+        """ % value, special_powers=True)
```

Comment:
Are you sure that 1 means 1 second? The value we set for the data layer is 20000 by default, so I assume that's ms.

Anyway, I'm fine with this, with one concern: any very fast value (probably even 20 seconds) is abusive to the remote IMAP server we check, to the point where they might actually blacklist us. 1 second would certainly be unacceptable.

I checked test/app code, and am troubled with how we currently do this. 

We set it to 20 seconds as soon as we init the Email object, then it stays that way for the whole test. Might stay that way for the whole test -run-, I'm not sure what does and doesn't survive if we do or don't restart. Certainly, if the test run aborts and leaves the phone on the email app, it will survive that and leave the phone fast-checking.

So while I think this change is basically OK, my specific concerns would be:

a) That we might want to set a higher minimum time as a bumper towards messing up too badly. We seem to only need 20 seconds, so maybe we should set it to 15 or 20, modulo comment below.

b) I think we should consider rewriting the test code more like:

      setup = SetupEmail()
      ...
      EmailUtil().send(...)
      setup.check_for_emails_interval(20000)
      try:
          ...
          # Wait for email notification
          system.wait_for_notification_toaster_displayed(timeout=60)
          system.wait_for_notification_toaster_not_displayed()
      finally:
           setup.check_for_emails_interval(300000)

...or something like that. That way it's only accelerated just long enough to get a fast sync for that one part of the test.

If we did that, I'd be up for dropping the sync rate (and minimum) way down to maybe 5 or 10 seconds, since it'd in theory only last one sync as long as the IMAP server processes the mail quickly. But either way, I think you really need to consider whether the current test leaves the phone hammering the server in the background.